src/lib.rs

```diff
@@ -513,8 +512,8 @@ pub mod private {
     pub use crate::opaque::Opaque;
     #[cfg(feature = "alloc")]
     pub use crate::result::{r#try, Result};
-    pub use crate::rust_option::assert_option_safe;
-    pub use crate::rust_option::RustOption;
+    pub use crate::rust_option::OptionFfi;
+    pub use crate::rust_option::OptionFfiInverse;
```

Comment:
so the struct RustOption no longer exists. Instead we now use the trait `OptionFfi` (implemented on `std::option::Option`) to link an FFI partner struct for Option<T> (for some T). This way the partner struct can be completely different (notably a different size) for different T. `OptionFfiInverse` is implemented on the partner struct as a helper to convert back to `std::option::Option`